lib/AST/ASTImporter.cpp

```diff
@@ -4392,31 +4404,39 @@ Decl *ASTNodeImporter::VisitClassTemplateDecl(ClassTemplateDecl *D) {
     for (unsigned I = 0, N = FoundDecls.size(); I != N; ++I) {
       if (!FoundDecls[I]->isInIdentifierNamespace(Decl::IDNS_Ordinary))
         continue;
-      
+
       Decl *Found = FoundDecls[I];
-      if (ClassTemplateDecl *FoundTemplate 
-                                        = dyn_cast<ClassTemplateDecl>(Found)) {
+      if (auto *FoundTemplate = dyn_cast<ClassTemplateDecl>(Found)) {
+
+        // We found a forward declaration but the class to be imported has a
+        // definition.
+        if (D->isThisDeclarationADefinition() &&
+            !FoundTemplate->isThisDeclarationADefinition()) {
+
+          // Lookup will find the fwd decl only if that is more recent than the
+          // definition. So, lets try to get the definition if that available
+          // in the redecl chain.
+          ClassTemplateDecl* TemplateWithDef = getDefinition(FoundTemplate);
```

Comment:
Does it make sense to call `!FoundTemplate->isThisDeclarationADefinition()` at all?